### DISCUSSION TITLE
Error since 0.18.0: "the AMP flag (--amp or SendPaymentRequest.Amp) must be set to pay an AMP invoice"

### DISCUSSION THREAD

**Discussion by danielalexiuc** [lightningnetwork/lnd#8917] [General]
I started getting this failure message on some outgoing payments since 0.18.2 (maybe since 0.18.0?). 

Currently, I don't set the 'amp' flag at all, however I do set the 'max_parts' field. This configuration has previously allowed me to pay AMP payments succcessfully.

I notice there were some changes around this recently, has it changed the default value for the 'amp' field to be false or something like that maybe?
https://github.com/lightningnetwork/lnd/issues/8728

**Comment by saubyk** [comment]
Hi @danielalexiuc yes, with 0.18, if you need to pay the amp invoice you need to set the `amp` flag to true. There was a bug previously which allowed amp invoices to be paid without setting that flag, although the flag was always there.

**Comment by danielalexiuc** [comment]
Thanks - I'm not sure I would call this a bug though, more like changing the default behaviour, was that intended?

The bug fix https://github.com/lightningnetwork/lnd/issues/8728 was to make it work when amp was explicitly set to true, but that fix also had the side-effect of changing the default behaviour of that paramter. This might be affecting a lot of attempted amp payments?

**Reply by saubyk** [reply]
so, just to be crystal clear here...the 'bug' that I am referring to is what you're referring to as "default behaviour", that's what was fixed. `amp` flag's intended objective was to provide a control, so that user is aware that they are paying an amp invoice (since amp invoice can be paid any number of time). But for some reason before 0.18 that flag was ignored for amp invoices, which has now been fixed.

The bug https://github.com/lightningnetwork/lnd/issues/8728 was an issue raised during 0.18 rc testing where there was a problem with paying amp invoices even after setting the flag. You should ignore this specific issue, as this issue only corrected a development mistake (to fix the original problem) which I described in the earlier paragraph

hope it's clear now

**Reply by danielalexiuc** [reply]
I see - thanks for the explanation. 👍
